Service migration

carlos-algms opened the pull request.

Commit:
temp commit

tcorreia-daitan committed:
overview and persona models

Commit:
overnight commit

Commit:
working weapon command

carlos-algms commented on integrations/battlelog/models/Overview.js:
```diff
@@ -0,0 +1,20 @@
+module.exports = class Overview {
+  constructor(overview) {
```
SE overview só contem as propriedades que você quer (nada a mais) você pode usar algo como `Object.assign(this, overview);`

talesprates replied:
pior que nao.. cada json desses eh gigantesco... estou pegando soh as properties que irei utilizar.. ai conforme a necessidade vou adicionando no modeo

carlos-algms commented on integrations/battlelog/overview.js:
```diff
@@ -1,4 +1,5 @@
 const request = require('request');
```
mudar de `request` para `axios` que é mais parecido com os Ajax de navegador

talesprates replied:
ok

carlos-algms commented on integrations/battlelog/weapon-stats/group.js:
```diff
@@ -0,0 +1,32 @@
+
+function reduceAllWeapons(wl1, wl2) {
+  const reducedList = {};
+  Object.keys(wl1).forEach((key) => {
```
aqui você pode usar o `Object.keys(wl1).reduce()` e dar return direto

talesprates replied:
arrumado

carlos-algms commented on integrations/chatbot/commands/weapon-tracking.js:
```diff
@@ -1,107 +1,44 @@
+function weaponTrack(message, callback, weaponSlug) {
+  Promise.all(
+    [
+      Promise.all(TRACKED_PLAYERS.map(personaId => weaponStats.getWeapons(personaId, weaponSlug))),
```
Acredito que você não precisa destes 2 `Promise.all()` aqui. 
Tenta remover, se não der, usa `Promise.resolve()` no lugar que é o mais correto.

carlos-algms commented on integrations/chatbot/commands/weapon-tracking.js:
```diff
@@ -1,107 +1,44 @@
-
-function generateServerMessage(playersWeapons, gun) {
+function generateResponseMessage(promises) {
   return new Promise((resolve) => {
```
Você pode usar o **destrutor** aqui assim `[ weapons, personas]` logo no cabeçalho da função mesmo e deletar as linhas 27 e 28. ;)

talesprates replied:
arrumado

carlos-algms commented on integrations/chatbot/commands/weapon-tracking.js:
```diff
@@ -1,107 +1,44 @@
+        .sort((weapon1, weapon2) => weapon2.kills - weapon1.kills);
+      return { weapon: sortedWeapons[TOPWEAPON], persona: playerPersona };
+    }).sort((player1, player2) => player2.weapon.kills - player1.weapon.kills)
+      .forEach((player, index) => {
```
este `forEach` pode ser substituído por outro `map` e a linha 29 pode vir na linha 31

talesprates replied:
feito

carlos-algms commented on integrations/chatbot/commands/weapon-tracking.js:
```diff
@@ -1,107 +1,44 @@
+      .forEach((player, index) => {
+        const accuracy = player.weapon.shotsHit / player.weapon.shotsFired;
+        serverMessage.push(`${index + 1}. **${player.persona.personaName}**`);
+        serverMessage.push(`\t*${player.weapon.slug.toUpperCase()}* - **${player.weapon.kills}** (*${accuracy.toFixed(2)}%*)`);
```
Porem aqui ao invés de push, será return

talesprates replied:
feito

carlos-algms commented:
cadê os testes unitários?

talesprates commented:
@carlos-algms feito o weapon tracking to jeito certo eu vou atras dos testes unitarios

tcorreia-daitan committed:
code review fixes

Commit:
implemented jasmine weapon tracking test

Commit:
migrated vehicle tracking commands and implemented jasmine tests

Commit:
overnight commit

Commit:
migrated overview tracking and implemented jasmine tests

Commit:
migrated server snapshot. TODO: snapshot mockups

Commit:
implemented jasmine tests for server tracking

Commit:
migration to services

overview and persona models

overnight commit
working weapon command
code review fixes

implemented jasmine weapon tracking test
migrated vehicle tracking commands and implemented jasmine tests
overnight commit
migrated overview tracking and implemented jasmine tests
migrated server snapshot. TODO: snapshot mockups
implemented jasmine tests for server tracking

Commit:
Merge remote-tracking branch 'origin/service-migration' into service-migration

carlos-algms commented on integrations/battlelog/models/Weapon.js:
```diff
@@ -0,0 +1,27 @@
+    this.timeEquipped = weapon.timeEquipped;
+  }
+
+  static add(w1, w2) {
```
`createInstance(....)` fica melhor

carlos-algms commented on integrations/battlelog/models/Weapon.js:
```diff
@@ -0,0 +1,27 @@
+  }
+
+  static add(w1, w2) {
+    const weapon = {};
```
desperdício de memória né. você pode dar um new e atribuir direto ao objeto final.
em todos os models. não precisa criar um objeto intermediário.

carlos-algms commented on integrations/battlelog/overview/group.js:
```diff
@@ -0,0 +1,15 @@
+};
+
+function getOverview(personaList) {
+  return new Promise((resolve, reject) =>
```
Não precisa de new Promise aqui retorna direto o Promise.all() que tá bala. 
Mesmo que `Overview.add` não seja Promise.

carlos-algms commented on integrations/battlelog/vehicle-stats/group.js:
```diff
@@ -0,0 +1,31 @@
+};
+
+function getAllVehicles(personaList) {
+  return new Promise((resolve, reject) => {
```
mesma coisa.

carlos-algms commented on integrations/battlelog/vehicle-stats/group.js:
```diff
@@ -0,0 +1,31 @@
+
+function reduceAllWeapons(vl1, vl2) {
+  return Object.keys(vl1).reduce((reducedList, key) => {
+    reducedList[key] = vl2[key] ? Vehicle.add(vl1[key], vl2[key]) : vl1[key];
```
isso aqui está um inferno pra ler ein. temos que pensar em algo melhor aqui.

carlos-algms commented on integrations/battlelog/vehicle-stats/group.js:
```diff
@@ -0,0 +1,31 @@
+}
+
+function getSingleVehicle(personaList, vehicleSlug) {
+  return new Promise((resolve, reject) => {
```
Não precisa de Promise

carlos-algms commented on integrations/battlelog/vehicle-stats/index.js:
```diff
@@ -0,0 +1,14 @@
+};
+
+function getVehicles(personaId, vehicleSlug) {
+  const individualOrGroup = Array.isArray(personaId) ? group : individual;
```
Já ví isso mais de uma vez, pode abstrair isso externamente.

carlos-algms commented on integrations/battlelog/vehicle-stats/individual.js:
```diff
@@ -0,0 +1,31 @@
+}
+
+function getSingleVehicle(personaId, vehicleSlug) {
+  return new Promise((resolve, reject) => {
```
- Promise

carlos-algms commented on integrations/battlelog/weapon-stats/group.js:
```diff
@@ -0,0 +1,31 @@
+};
+
+function getAllWeapons(personaList) {
+  return new Promise((resolve, reject) => {
```
- Promise

carlos-algms reviewed (commented):
Os mocks parecem bons, não tive tempo de rever os specs. tá indo bem.
